include/revng/Model/CallEdge.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "revng/Model/FunctionEdge.h"
+#include "revng/Model/FunctionEdgeType.h"
+#include "revng/Model/GeneratedCallEdge.h"
+
+/// A CFG edge to represent function calls (direct, indirect and tail calls)
+class model::CallEdge : public model::generated::CallEdge {
+public:
+  using generated::CallEdge::CallEdge;
+
+public:
+  static bool classof(const FunctionEdge *A) {
+    return FunctionEdgeType::isCall(A->Type);
+  }
+
+public:
+  bool verify() const debug_function;
+  bool verify(bool Assert) const debug_function;
+  bool verify(VerifyHelper &VH) const;
+  void dump() const debug_function;
+
+public:
+  // TODO: ideally we should not need to define this here
+  static constexpr const char *Tag = "!CallEdge";
```

Comment:
Didn't we decide for autogenerating a `getTag` trait for this?

Reply:
Actually, after we changed the way we handle inheritance, there's no need anymore, it can directly be defined in the autogenerated classes :)